File: starters/spring/src/main/java/com/enistere/core/platform/persistence/PersistencePort.java
package com.enistere.core.platform.persistence;

import java.util.Optional;

/**
 * Domain-neutral persistence seam. Capabilities specialize it with their own
 * domain types; Spring Data and JPA remain adapter details.
 */
public interface PersistencePort<Entity, Identifier> {
    Optional<Entity> findById(Identifier id);
    Entity save(Entity entity);
    void deleteById(Identifier id);
}

File: starters/nestjs/src/platform/persistence/persistence.port.ts
/**
 * Domain-neutral persistence seam.
 *
 * Capabilities specialize this port with their own entity and identifier types.
 * The contract intentionally exposes no Prisma type: Prisma belongs to adapters,
 * never to domain or capability contracts.
 */
export interface PersistencePort<Entity, Identifier> {
  findById(id: Identifier): Promise<Entity | null>;
  save(entity: Entity): Promise<Entity>;
  deleteById(id: Identifier): Promise<void>;
}
